src/system/array.ts

```diff
@@ -85,7 +85,7 @@ export namespace Arrays {
         value?: T;
 
         // parent?: IHierarchicalItem<T>;
-        children: { [key: string]: IHierarchicalItem<T> } | undefined;
+        children: Map<string, IHierarchicalItem<T>> | undefined;
```

Comment:
I kept the "`| undefined`" here, but it seems to me that we could rewrite the code easily to use an empty map instead (or an empty object before). Same for `descendants`.
Is there a good reason ?

Reply:
Mainly for performance, to avoid creating objects that will need to be GC'd that we won't need.